### PR TITLE
GraphInferenceEngine support

### PR DESCRIPTION
**Pull request by poedator**
enabling GraphInferenceEngine use in non-offloading setting. Consistent with [Sequoia testbed code](https://github.com/Infini-AI-Lab/Sequoia/blob/df58fc0408867e7cf5397a2dd88ce8a525b93695/tests/testbed.py#L248).

### CODE REVIEW

File: benchmark_inference.py [hunk 1]
```diff
@@ -37,28 +37,39 @@ def benchmark(args):
...
+                               attn_mask=attn_mask[..., :args.prefix_length,:])
+        print("test run OK")
 
     avg_forward_pass_times = []
     for decode_length in decode_lengths:
         input_ids = torch.randint(low=3, high=30000, size=(1, decode_length), device=device)
         storage_ids = torch.arange(decode_length, device=device) + args.prefix_length
         position_ids = storage_ids.clone().unsqueeze(0)
-        curr_attn_mask = attn_mask[..., args.prefix_length: args.prefix_length + decode_length,:args.prefix_length + decode_length].clone()
+        if args.offloading:
```
Comment (avnermay) [inline benchmark_inference.py]: why does the attn_mask behavior throughout this PR depend on offloading vs. not offloading?

File: benchmark_inference.py [hunk 2]
```diff
@@ -37,28 +37,39 @@ def benchmark(args):
...
         graph_engine = OffloadEngine(max_length=args.max_length, model_name_or_path=args.model, dtype=dtype, device=device)  # set stay_layers?
+        graph_engine.inference(input_ids=prefix, storage_ids=prefix_storage_ids, position_ids=prefix_position_ids, \
+                               attn_mask=attn_mask[..., :args.prefix_length,:args.prefix_length])
     else:
-        graph_engine = GraphInferenceEngineTG(max_length=args.max_length, model_name_or_path=args.model, dtype=dtype, device=device, offloading=args.offloading)
-
-    graph_engine.inference(input_ids=prefix, storage_ids=prefix_storage_ids, position_ids=prefix_position_ids, attn_mask=attn_mask[..., :args.prefix_length,:args.prefix_length])
+        graph_engine = GraphInferenceEngine(max_length=args.max_length, model_name_or_path=args.model, dtype=dtype, device=device)
+        print("initializing GraphInferenceEngine model")
+        graph_capture_list = [1,2,4,8,16,32,64,128,256,512,768,1024]
```
Comment (avnermay) [inline benchmark_inference.py]: can you not hardcode this?  use decode_lengths

File: benchmark_inference.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import time
 import torch
 
-from Engine.Engine import GraphInferenceEngineTG
+from Engine.Engine import GraphInferenceEngineTG, GraphInferenceEngine
```
Comment (avnermay) [inline benchmark_inference.py]: We no longer need  to import GraphInferenceEngineTG

File: benchmark_inference.py [hunk 4]
```diff
@@ -37,28 +37,39 @@ def benchmark(args):
...
+            if args.offloading:
+                graph_engine.set_kv_len(args.prefix_length)
 
         torch.cuda.synchronize()
         t1 = time.time()
 
         for _ in trange(args.num_repeats, desc=f"measuring, {decode_length=}", leave=False):
             graph_engine.inference(input_ids=input_ids, storage_ids=storage_ids, position_ids=position_ids, attn_mask=curr_attn_mask)
-            graph_engine.set_kv_len(args.prefix_length)
+            if args.offloading:
```
Comment (avnermay) [inline benchmark_inference.py]: why is calling set_kv_len dependent on offloading?

**Review by avnermay — CHANGES_REQUESTED** [review]
Thanks Ruslan! Left some comments.

File: benchmark_inference.py (hunk 1 above)
Comment (poedator) [reply benchmark_inference.py]: the two classes have different approaches to the AM. One needs just enough width to cover the input, the other has  fixed "window" for context and new tokens (to make it possible to compile).

File: benchmark_inference.py (hunk 4 above)
Comment (poedator) [reply benchmark_inference.py]: it depends on the class. Fixed the condition accordingly

**Comment by poedator** [comment]
Now that @avnermay suggestions are all addressed, I merge this PR

**Pull request merged**